app/hooks/StrapiAPI.ts

```diff
@@ -0,0 +1,32 @@
+import useSWR from "swr";
+import fetcher from "utils/fetcher";
+import { type Footer, type StrapiResponse } from "models/Strapi";
+import config from "../config";
+
+interface SWRHookResult<T> {
+  data: T;
+  error?: Error;
+  isLoading: boolean;
+}
+
+function useStrapiHook<T>(path: string): SWRHookResult<T> {
+  const dataFetcher = () =>
+    fetcher<StrapiResponse<T>>(`${config.STRAPI_API_URL}/api/${path}`, {
+      Authorization: `Bearer ${config.STRAPI_API_TOKEN}`,
+    });
+
+  const { data, error, isLoading } = useSWR<StrapiResponse<T>>(
+    `/api/${path}`,
+    dataFetcher
+  );
+
+  return {
+    data: data?.data ? (data.data as any).attributes : null,
```

Comment:
Huh, I'm surprised TS doesn't infer the type of `data.data` here so you wouldn't have to use `any`. Or is this related to the type for `data` on line 7 maybe needing to be a union of `T | null` (which I think is possible because strapi returns `data = null` for an error response, right?). Just eye-balling here...

Reply:
Hey @rosschapman , you are correct  and updating line 7 to be a union removes the need to specify any.